### PR TITLE
[docs] Update `LocalizationProvider` `dateAdapter` with a link to doc

### PR DESCRIPTION
**Pull request by LukasTy**
Created [from](https://mui-org.slack.com/archives/C041SDSF32L/p1675893700633939):
> What does DateAdapter mean, where do I find it, what are the options?
> sent from https://mui.com/x/api/date-pickers/localization-provider/

- Support parsing `@see` with `{@link}` JSDoc into an anchor
- Update `LocalizationProvider` `dateAdapter` JSDoc to no longer refer to `DateIO` and link to a doc

JSDoc example in IDE:
Screenshot 2023-02-09 at 15 22 16

Rendered [doc example](https://deploy-preview-7872--material-ui-x.netlify.app/x/api/date-pickers/localization-provider/#:~:text=you%20are%20using-,dateAdapter,-func).

### CODE REVIEW

File: packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx [hunk 1]
```diff
@@ -28,7 +28,10 @@ if (process.env.NODE_ENV !== 'production') {
 
 export interface LocalizationProviderProps<TDate> {
   children?: React.ReactNode;
-  /** DateIO adapter class function */
+  /**
+   * Date library adapter class function.
+   * @see See the {@link https://next.mui.com/x/react-date-pickers/adapters-locale/#set-a-custom-locale available adapters page} for more details.
```
Comment (alexfauquette) [inline packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx:33]: The page you are pointing to explains how to use `adapterLocale`, not the `dateAdapter`
```suggestion
   * @see See the {@link https://next.mui.com/x/react-date-pickers/#date-library available adapters page} for more details.
```

Comment (LukasTy) [reply packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx:33]: I was thinking more about the examples of adapters usage and that page you linked has no examples...
Maybe the way could be to add a section referencing my mentioned page in your suggested one to sort of link them together. 🤔

Comment (alexfauquette) [reply packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx:33]: For a docs example, you have the following link which I an interactive demo: https://next.mui.com/x/react-date-pickers/getting-started/#code-setup

The problem with the current link is that you land on a screen where everything is dedicated to dayjs, so my first idea was

> Ok I will not find the information if I want to use luxon, or another adapter

image

Comment (LukasTy) [reply packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx:33]: > The problem with the current link is that you land on a screen where everything is dedicated to dayjs, so my first idea was
> 
> > Ok I will not find the information if I want to use luxon, or another adapter

Fair point, it's not the best page to link to in order to have the best initial visibility, but actually so are other options. 🙈 By the way, the sidebar links do have all the adapter libraries listed. 🙈 



> For a docs example, you have the following link which I an interactive demo: https://next.mui.com/x/react-date-pickers/getting-started/#code-setup

I actually think that this is the best link we can have for `dateAdapter` doc, wouldn't you agree? 🤔

**Review by alexfauquette — APPROVED** [review]
Amazing feature 👌🎉

**Pull request merged**